File: services/payment/src/main/java/com/example/iter/payment/dto/response/PaymentReadyResponse.java
package com.example.iter.payment.dto.response;

import com.example.iter.reservation.api.RentalInfo;

import java.math.BigDecimal;

public record PaymentReadyResponse(
        Long rentalId,
        String orderId,
        String orderName,
        BigDecimal amount,
        String clientKey,
        String customerKey
) {
    public static PaymentReadyResponse of( RentalInfo rental, String orderId, BigDecimal amount, String clientKey ) {
        return new PaymentReadyResponse(
                rental.rentalId(),
                orderId,
                rental.productName(),
                amount,
                clientKey,
                "user-" + rental.renterId()
        );
    }

}

File: services/payment/src/main/kotlin/com/example/iter/payment/dto/response/PaymentReadyResponse.kt
package com.example.iter.payment.dto.response

import com.example.iter.reservation.api.RentalInfo
import java.math.BigDecimal

@JvmRecord
data class PaymentReadyResponse(
    val rentalId: Long,
    val orderId: String,
    val orderName: String,
    val amount: BigDecimal,
    val clientKey: String,
    val customerKey: String,
) {
    companion object {
        @JvmStatic
        fun of(rental: RentalInfo, orderId: String, amount: BigDecimal, clientKey: String): PaymentReadyResponse =
            PaymentReadyResponse(
                rental.rentalId,
                orderId,
                rental.productName,
                amount,
                clientKey,
                "user-" + rental.renterId,
            )
    }
}
